test/test_images/auditlogs_target/main.go

```diff
@@ -63,16 +65,22 @@ func main() {
 	timer := time.NewTimer(time.Second * time.Duration(duration))
 	defer timer.Stop()
 	go func() {
-		<-timer.C
-		// Write the termination message if time out
-		fmt.Printf("time out to wait for event with type %q source %q subject %q service_name %q method_name %q resource_name %q .\n",
-			r.Type, r.Source, r.Subject, r.ServiceName, r.MethodName, r.ResourceName)
-		if err := r.writeTerminationMessage(map[string]interface{}{
-			"success": false,
-		}); err != nil {
-			fmt.Printf("failed to write termination message, %s.\n", err.Error())
+		select {
+		case <-shutDown:
+			// Give the receiver a little time to finish responding.
+			time.Sleep(time.Second)
+			os.Exit(0)
+		case <-timer.C:
```

Comment:
Rather than use a timer and shutdown channel, why not just attach a timeout to the receiver context and communicate the result back to the main thread with a channel? The main thread can then cleanup and exit normally without needing to call os.Exit from a goroutine.

Reply:
Done. 

But this assumes that the server shuts down gracefully when the context is closed. I looked at the code and it seems to today, but I don't think that is guaranteed in the future.